src/System.Memory/src/System/Buffers/SpanLiteralExtensions.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Buffers
+{
+    internal class SpanLiteralExtensions
+    {
+        internal static void AppendAsLiteral(ReadOnlySpan<byte> span, StringBuilder sb)
```

Comment:
Would it make sense to share diagnostics formatting between span, memory, and buffer?

@ahsonkhan?

Reply:
@KrzysztofCwalina, what do you mean by diagnostics formatting?